packages/houdini/src/codegen/generators/artifacts/artifacts.test.ts

```diff
@@ -869,6 +869,293 @@ describe('mutation artifacts', function () {
 		`)
 	})
 
+	test('insert operation allList', async function () {
+		const mutationDocs = [
+			mockCollectedDoc(
+				`mutation A {
+					addFriend {
+						friend {
+							...All_Users_insert @allList
+						}
+					}
+				}`
+			),
+			mockCollectedDoc(
+				`query TestQuery {
+					users(stringValue: "foo") @list(name: "All_Users") {
+						firstName
+					}
+				}`
+			),
+		]
+
+		// execute the generator
+		await runPipeline(config, mutationDocs)
+
+		// load the contents of the file
+		const queryContents = await fs.readFile(
+			path.join(config.artifactPath(mutationDocs[0].document))
+		)
+		expect(queryContents).toBeTruthy()
+		// parse the contents
+		const parsedQuery: ProgramKind = recast.parse(queryContents!, {
+			parser: typeScriptParser,
+		}).program
+		// verify contents
+		expect(parsedQuery).toMatchInlineSnapshot(`
```

Comment:
This test can be simplified  using the custom matcher. The test you copied from was an old one 😅 

```suggestion

		expect(mutationDocs[0]).toMatchInlineSnapshot(`
```